packages/components/src/internal/components/chart/utils.ts

```diff
@@ -49,6 +50,101 @@ export function processChartData(response: ISelectRowsResult, options?: ProcessC
     return { barFillColors, data };
 }
 
+interface PercentageBarProps {
+    queryKey: string,
+    name: string;
+    label: string,
+    useForSubtitle?: boolean,
+    className?: string,
+    appURL?: AppURL,
+    filled?: boolean,
+}
+
+export function createPercentageBarData(
+    row: Record<string, any>,
+    itemNounPlural: string,
+    unusedLabel: string,
+    totalCountKey: string,
+    percentageBars: PercentageBarProps[],
+    baseAppURL?: AppURL,
+    urlFilterKey?: string
+) : { data: HorizontalBarData[], subtitle: string } {
+    const totalCount = caseInsensitive(row, totalCountKey)?.value ?? 0;
+    let unusedCount = totalCount;
+    const data = [];
+    let subtitle;
+    if (totalCount > 0) {
+        percentageBars.forEach(percentageBar => {
+            const itemCount = caseInsensitive(row, percentageBar.queryKey)?.value ?? 0;
+            unusedCount = unusedCount - itemCount;
+            const itemPct = (itemCount/totalCount) * 100;
+            const title = `${itemCount.toLocaleString()} of ${totalCount.toLocaleString()} ${itemNounPlural.toLowerCase()} are ${percentageBar.label.toLowerCase()}`;
+
+            data.push({
+                title,
+                name: percentageBar.name,
+                className: percentageBar.className,
+                count: itemCount,
+                totalCount: totalCount,
+                percent: itemPct,
+                href: percentageBar.appURL?.toHref() ?? baseAppURL?.addFilters(Filter.create(urlFilterKey, percentageBar.label)).toHref(),
+                filled: percentageBar.filled ?? true,
+            });
+
+            if (percentageBar.useForSubtitle) {
+                subtitle = `${title} (${(itemPct > 0 && itemPct < 1 ? '< 1' :  Math.trunc(itemPct))}%)`;
+            }
+        });
+
+        if (unusedLabel) {
```

Comment:
```suggestion
        if (unusedLabel && unusedCount) {
```
Do we need data if all are used?

Reply:
Good call. Done.